config/config_test.go

```diff
@@ -1321,7 +1317,6 @@ func TestCheckExchangeConfigValues(t *testing.T) {
 
 	// Test API settings migration
```

Comment:
Can you check if that's true?
Does another process have test.json open?
I can't imagine that this is go itself, because it wouldn't constitute "another process" itself.

Reply:
No idea seems like its a windows issue, happening on another computer:

```
    c:\Users\graem\ryan\thrasher-corp\gocryptotrader\config\config_test.go:2105: migrateConfig() error = remove test.json: The process cannot access the file because it is being used by another process., wantErr false
=== RUN   TestMigrateConfig/source_same_as_target
=== RUN   TestMigrateConfig/source_and_target_present
config file already found in 'C:\Users\graem\AppData\Local\Temp\TestMigrateConfig1792913445\001\config.json'; not overwriting, defaulting to C:\Users\graem\AppData\Local\Temp\TestMigrateConfig1792913445\001\src\config.dat
    c:\Users\graem\ryan\thrasher-corp\gocryptotrader\config\testing.go:1232: TempDir RemoveAll cleanup: remove C:\Users\graem\AppData\Local\Temp\TestMigrateConfig1792913445\001\config.json: The process cannot access the file because it is being used by another process.
--- FAIL: TestMigrateConfig (1.87s)
    --- PASS: TestMigrateConfig/nonexisting (0.00s)
    --- FAIL: TestMigrateConfig/source_present,_no_target_dir (0.00s)
    --- PASS: TestMigrateConfig/source_same_as_target (0.00s)
    --- PASS: TestMigrateConfig/source_and_target_present (0.00s)
```

Reply:
There was an inconsistency on test.json being not in TempDir, and trying to remove it when it shouldn't have been there.
I'm not sure if I've fixed it or not, but let's see.
Since I haven't been able to reproduce, Could you please test this for me?

Fixed 9a59bae7f

Reply:
See [this](https://github.com/thrasher-corp/gocryptotrader/pull/1671#discussion_r1818248800)